Composer/packages/tools/language-servers/language-understanding/src/luParser.ts

```diff
@@ -0,0 +1,125 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import { fork, ChildProcess } from 'child_process';
+import path from 'path';
+
+import { parser } from '@microsoft/bf-lu/lib/parser';
+import { updateIntent, checkSection } from '@bfc/indexers/lib/utils/luUtil';
+import { luIndexer } from '@bfc/indexers';
+import { LuFile, ILUFeaturesConfig, LuIntentSection, Diagnostic } from '@bfc/shared';
+import uniqueId from 'lodash/uniqueId';
+
+export const isTest = process.env?.NODE_ENV === 'test';
+export interface WorkerMsg {
+  id: string;
+  type: 'parse' | 'updateIntent' | 'parseFile' | 'checkSection' | 'getFoldingRanges';
+  error?: any;
+  payload?: any;
+}
+
+export class LuParserWithoutWorker {
+  public async parse(content: string, id = '', config: ILUFeaturesConfig): Promise<LuFile> {
+    return luIndexer.parse(content, id, config);
+  }
+  public async updateIntent(
+    luFile: LuFile,
+    intentName: string,
+    intent: { Name?: string; Body?: string } | null,
+    luFeatures: ILUFeaturesConfig
+  ): Promise<LuFile> {
+    return updateIntent(luFile, intentName, intent, luFeatures);
+  }
+  public async parseFile(text, log, locale): Promise<any> {
```

Comment:
is the parsed result just string instead of any?

Reply:
The parse result is complex, but the bf-lu library doesn't have a type file to export.

Reply:
I will sync with fei about this one